Library/Homebrew/link.rb

```diff
@@ -0,0 +1,62 @@
+def link_src_dst_dirs(src_dir, dst_dir, command, link_dir: false)
+  return unless src_dir.exist?
+  conflicts = []
+  src_paths = link_dir ? [src_dir] : src_dir.find
+  src_paths.each do |src|
+    next if src.directory? && !link_dir
+    dst = dst_dir/src.relative_path_from(src_dir)
+    if dst.symlink?
+      next if src == dst.resolved_path
+      dst.unlink
+    end
+    if dst.exist?
+      conflicts << dst
+      next
+    end
+    dst_dir.parent.mkpath
+    dst.make_relative_symlink(src)
+  end
+
+  return if conflicts.empty?
+  onoe <<-EOS.undent
+    Could not link:
+    #{conflicts.join("\n")}
+
+    Please delete these paths and run `#{command}`.
+  EOS
+end
+
+def unlink_src_dst_dirs(src_dir, dst_dir, unlink_dir: false)
+  return unless src_dir.exist?
+  src_paths = unlink_dir ? [src_dir] : src_dir.find
+  src_paths.each do |src|
+    next if src.directory? && !unlink_dir
+    dst = dst_dir/src.relative_path_from(src_dir)
+    dst.delete if dst.symlink? && src == dst.resolved_path
+    dst.parent.rmdir_if_possible
+  end
+end
+
+def link_manpages(path, command)
```

Comment:
I think it makes enough sense as-is to leave it and only update the implementation if a situation presents that other man sections are needed. Adding manX would likely mean reorganizing a bunch of folders across various repos, so it would be a fairly heavy change that isn't just a matter of adding a new method